src/test/scala/collection/immutable/TestLazyListExtensions.scala

```diff
@@ -0,0 +1,81 @@
+/*
+ * Scala (https://www.scala-lang.org)
+ *
+ * Copyright EPFL and Lightbend, Inc.
+ *
+ * Licensed under Apache License 2.0
+ * (http://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package scala.collection.immutable
+
+import org.junit.Assert._
+import org.junit.Test
+
+import LazyListExtensions._
+
+class TestLazyListExtensions {
+  @Test
+  def testEmpty1(): Unit = {
+    val xs = LazyList.empty  // realized
+    val cyc = xs.cycle
+    assertTrue(cyc.isEmpty)
+    assertTrue(cyc.size == 0)
+    assertEquals(Nil, cyc.toList)
+  }
+  @Test
+  def testEmpty2(): Unit = {
+    val xs = LazyList()  // not realized
+    val cyc = xs.cycle
+    assertTrue(cyc.isEmpty)
+    assertTrue(cyc.size == 0)
+    assertEquals(Nil, cyc.toList)
+  }
+  @Test
+  def testNonEmpty(): Unit = {
+    val xs = LazyList(1, 2, 3)
+    val cyc = xs.cycle
+    assertFalse(cyc.isEmpty)
+    assertEquals(LazyList(1, 2, 3, 1, 2, 3, 1, 2), cyc.take(8))
+  }
+  @Test
+  def testToString(): Unit = {
+    assertEquals("LazyList()",
+      LazyList.empty.cycle.toString)
+    assertEquals("LazyList(<not computed>)",
+      LazyList(1, 2, 3).cycle.toString)
+    // note cycle detection here!
+    assertEquals("LazyList(1, 2, 3, <cycle>)",
+      LazyList(1, 2, 3).cycle.force.toString)
+  }
+  @Test
+  def testRepeats(): Unit = {
+    val xs = LazyList(1, 2, 3)
+    val cyc = xs.cycle
+    assertFalse(cyc.isEmpty)
+    assertEquals(LazyList(1, 2, 3, 1, 2, 3, 1, 2), cyc.take(8))
+  }
+  @Test
+  def testConstantMemory1(): Unit = {
+    val xs = LazyList(1, 2, 3)
+    val cyc = xs.cycle
+    assertTrue(cyc.tail eq cyc.tail.tail.tail.tail)
+    assertTrue(cyc.tail eq cyc.drop(4))   // TODO: fails
```

Comment:
the vast majority of methods on `LazyList` (`drop` included) do not return cyclic `LazyList`s when called on cyclic `LazyList`s (see https://github.com/scala/bug/issues/11270#issuecomment-452515304 and https://github.com/scala/scala/pull/7692#issuecomment-458718024)

Reply:
actually, I take that back. I feel like `drop` *should* return a cyclic `LazyList`